BatteryStateManagement_proj.c

```diff
@@ -0,0 +1,209 @@
+ * 
+ *     \param       NA 
+ *     \returns     NA
+ *
+*//*------------------------------------------------------------------------*/
+void batteryWarnHandlingMinRange_v()
+{
+ static float range;
+ /*warning level percentage calulcator : 5% of max range*/
+ 
+  range = (BattParmt_str_p->minRange +  BattParmt_str_p->warnLevel);
+ 
+   /*warning prediction min - Approaching discharge*/
+    if ( (BattParmt_str_p->batteryParameter > BattParmt_str_p->minRange) &&  (BattParmt_str_p->batteryParameter <= range))
+    {
+     printf("Warn---> %s---->Entering Min\n",batPar[BattParmt_str_p->batParIndex]);
+    }
+}
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryStateValidation_i
+ */
+/*!    \brief       Monitoring the battery states
+ * 
+ *     \param       temperature,soc,chargeRate
+ *     \returns     batterystate -OK/NOK
+ *
+*//*------------------------------------------------------------------------*/
+int batteryStateValidation_i(float temperature, float soc, float chargeRate)
+{ 
+ 
+ int retTempStat_i, retsocStat_i,retchargeStat_i;
+ 
+  if ( GERMAN_LANG_SLECTD != langSelected_uint)
+  {
+  /*Validation done with current battery parameter,Min and max range in english lang*/
+   
+  BattParmt_str_p->batteryParameter =temperature;
+  BattParmt_str_p->minRange = 0;
+  BattParmt_str_p->maxRange =45;
+  BattParmt_str_p->batParIndex =0;
+  batteryWarnHandling_v();
+  retTempStat_i   = batteryCondMonitor_i();
+   
+  BattParmt_str_p->batteryParameter =soc;
+  BattParmt_str_p->minRange = 20;
+  BattParmt_str_p->maxRange =80;
+  BattParmt_str_p->batParIndex =1;
+  batteryWarnHandling_v();
+  retsocStat_i    = batteryCondMonitor_i();
+   
+  BattParmt_str_p->batteryParameter =chargeRate;
+  BattParmt_str_p->minRange = 0;
+  BattParmt_str_p->maxRange =0.8;
+  BattParmt_str_p->batParIndex =2;
+  batteryWarnHandling_v(); 
+  retchargeStat_i = batteryCondMonitor_i();
+   
+  }
+  else
+  {
+  /*Validation done with cureent battery parameter,Min and max range in german lang*/
+  BattParmt_str_p->batteryParameter =temperature;
+  BattParmt_str_p->minRange = 0;
+  BattParmt_str_p->maxRange =45;
+  BattParmt_str_p->batParIndex =3;
+  batteryWarnHandling_v(); 
+  retTempStat_i   = batteryCondMonitor_i();
+   
+  BattParmt_str_p->batteryParameter =soc;
+  BattParmt_str_p->minRange = 20;
+  BattParmt_str_p->maxRange =80;
+  BattParmt_str_p->batParIndex =4;
+  batteryWarnHandling_v();
+  retsocStat_i    = batteryCondMonitor_i();
+   
+  BattParmt_str_p->batteryParameter =chargeRate;
+  BattParmt_str_p->minRange = 0;
+  BattParmt_str_p->maxRange =0.8;
+  BattParmt_str_p->batParIndex =5; 
+  batteryWarnHandling_v(); 
+  retchargeStat_i = batteryCondMonitor_i();
+  }
+  /*return battery state ok /nok*/
+  return ((retTempStat_i & retsocStat_i) & retchargeStat_i);
+}
+
+int main() 
+{
+ /*Initilizing the ptr */
+  BattParmt_str_p = &BattParmt_strPtr_s;
```

Comment:
Why can't we use `BattParmt_strPtr_s` directly? Why is a pointer initialized in addition?